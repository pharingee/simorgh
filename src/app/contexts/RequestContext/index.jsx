import React from 'react';
import { bool, node, oneOf, string } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';

export const RequestContext = React.createContext('default');

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  previousPath,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    platform,
    statsDestination,
    statsPageIdentifier,
    previousPath,
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf(['article', 'frontPage', 'media']).isRequired,
  service: string.isRequired,
  previousPath: string,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  previousPath: null,
};
