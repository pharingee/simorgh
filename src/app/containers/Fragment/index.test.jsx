import React from 'react';
import { shouldMatchSnapshot } from '../../../testHelpers';
import Fragment from './index';
import { ServiceContext } from '../../contexts/ServiceContext';

const CreateFragment = (newsServiceContextStub, text, attributes) => (
  <ServiceContext.Provider value={newsServiceContextStub}>
    <Fragment text={text} attributes={attributes} />
  </ServiceContext.Provider>
);

const newsContext = {
  service: 'news',
};

const persianContext = {
  service: 'persian',
};

describe('Fragment', () => {
  describe('with no attributes', () => {
    shouldMatchSnapshot(
      'should render just text',
      CreateFragment(newsContext, 'This is some text with no attributes', []),
    );
  });

  describe('with bold attributes', () => {
    shouldMatchSnapshot(
      'should render text wrapped in a bold DOM element',
      CreateFragment(newsContext, 'This is some text with bold attributes', [
        'bold',
      ]),
    );
  });

  describe('with italic attributes', () => {
    shouldMatchSnapshot(
      'should render text wrapped in an italic DOM element',
      CreateFragment(newsContext, 'This is some text with italic attributes', [
        'italic',
      ]),
    );
  });

  describe('with bold and italic attributes', () => {
    shouldMatchSnapshot(
      'should render text wrapped in bold and italic DOM elements',
      CreateFragment(
        newsContext,
        'This is some text with bold and italic attributes',
        ['bold', 'italic'],
      ),
    );
  });

  describe('with an unknown attribute', () => {
    shouldMatchSnapshot(
      'should ignore the attribute',
      CreateFragment(
        newsContext,
        'This is some text with a bold and unknown attribute',
        ['bold', 'unknown'],
      ),
    );
  });

  describe('with italic attributes in Farsi', () => {
    shouldMatchSnapshot(
      'should render text wrapped in an italic DOM element',
      CreateFragment(
        persianContext,
        'This is some text with italic attributes',
        ['italic'],
      ),
    );
  });
});
