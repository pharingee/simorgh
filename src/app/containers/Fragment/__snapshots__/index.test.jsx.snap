// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment with an unknown attribute should ignore the attribute 1`] = `
<b>
  This is some text with a bold and unknown attribute
</b>
`;

exports[`Fragment with bold and italic attributes should render text wrapped in bold and italic DOM elements 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  font-family: inherit;
  font-weight: inherit;
}

<i
  className="c0"
>
  <b>
    This is some text with bold and italic attributes
  </b>
</i>
`;

exports[`Fragment with bold attributes should render text wrapped in a bold DOM element 1`] = `
<b>
  This is some text with bold attributes
</b>
`;

exports[`Fragment with italic attributes in Farsi should render text wrapped in an italic DOM element 1`] = `
.c0 {
  font-family: NassimPersian,Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-family: inherit;
  font-weight: inherit;
}

<i
  className="c0"
>
  This is some text with italic attributes
</i>
`;

exports[`Fragment with italic attributes should render text wrapped in an italic DOM element 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  font-family: inherit;
  font-weight: inherit;
}

<i
  className="c0"
>
  This is some text with italic attributes
</i>
`;

exports[`Fragment with no attributes should render just text 1`] = `"This is some text with no attributes"`;
